core/trino-main/src/main/java/io/trino/sql/planner/planprinter/DistributedFragmentRepresentation.java

```diff
@@ -0,0 +1,495 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.planprinter;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import io.trino.cost.PlanCostEstimate;
+import io.trino.cost.PlanNodeStatsAndCostSummary;
+import io.trino.cost.PlanNodeStatsEstimate;
+import io.trino.sql.planner.Symbol;
+import io.trino.sql.planner.planprinter.NodeRepresentation.TypedSymbol;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.sql.planner.planprinter.RendererUtils.formatAsCpuCost;
+import static io.trino.sql.planner.planprinter.RendererUtils.formatAsDataSize;
+import static io.trino.sql.planner.planprinter.RendererUtils.formatAsLong;
+import static io.trino.sql.planner.planprinter.RendererUtils.translateOperatorTypes;
+import static java.util.Collections.emptyMap;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+public class DistributedFragmentRepresentation
+{
+    private final String name;
+    private final String identifier;
+    private final List<TypedSymbol> outputs;
+    private final String details;
+    private final DistributedFragmentStats distributedFragmentStats;
+    private final Optional<PlanNodeStatsAndCostSummary> planNodeStatsAndCostSummary;
```

Comment:
Are these the stats associated with `nodeRepresentation` below?

Reply:
Yes, these are from `NodeRepresentation` if the `PlanNodeStats` is of type `WindoPlanNodeStats`. Hence, declared these as Optional.